### PR TITLE
Feat: number input

### PR DESCRIPTION
**Pull request by mjmonline**

### CODE REVIEW

File: src/Molecules/Input/Number/Number.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+    defaultValue = 0,
+    value: controlledValue,
+    fieldId,
+    intl,
+    size,
+    step = 1,
+    min = 0,
+    max,
+    name,
+    noSteppers,
```
Comment (jesperwiner) [inline src/Molecules/Input/Number/Number.tsx]: I prefer to have props that activate instead of turning off a feature

**Comment by Maxenboy** [comment]
Would be nice to have autoFocus prop

File: src/Molecules/Input/Number/Number.tsx (hunk 1 above)
Comment (mjmonline) [reply src/Molecules/Input/Number/Number.tsx]: We talked about this but for others wondering, the steppers are there by default even natively

File: src/Molecules/Input/Number/Number.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        border-color: ${p.theme.color.inputBorderHover};
+        z-index: 2;
+      }
+`}
+`;
+
+const focusBorderStyles = css<Pick<Props, 'error'>>`
+  &:focus {
+    border-color: ${p =>
+      hasError(p.error) ? p.theme.color.inputBorderError : p.theme.color.borderActive};
```
Comment (shmidt-i) [inline src/Molecules/Input/Number/Number.tsx]: always p.theme.color.borderActive

File: src/Molecules/Input/Number/Number.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+  const inputRef = useRef<HTMLInputElement>(null);
+  const isControlled = controlledValue && controlledValue >= 0;
+  const sanitized = {
+    max: max ? getStringAsNumber(max) : undefined,
+    min: min ? getStringAsNumber(min) : undefined,
+    step: isNumber(step) ? step : getStringAsNumber(step),
+    uncontrolledValue: getStringAsNumber(uncontrolledValue),
+  };
+
+  const getUpdateValue = (increment: boolean) => {
```
Comment (shmidt-i) [inline src/Molecules/Input/Number/Number.tsx]: React.useCallback maybe

File: src/Molecules/Input/Number/Number.tsx [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+            onKeyUp,
+            ref: inputRef,
+            required,
+            step,
+            success,
+            value: controlledValue || uncontrolledValue,
+          }}
+          {...(hasError(error) ? { 'aria-invalid': true } : {})}
+        />
+        {!noSteppers && (
```
Comment (shmidt-i) [inline src/Molecules/Input/Number/Number.tsx]: can merge with next if statement

File: src/Molecules/Input/Number/Number.types.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+   * (better in pixels), because
+   * that will affect wrapping
+   * of the error/info text
+   * underneath
+   */
+  width?: string | number;
+
+  onStepUp?: (() => void) | ((value?: string) => void);
+  onStepDown?: (() => void) | ((value?: string) => void);
+  onChange?: React.ChangeEventHandler<HTMLInputElement>;
```
Comment (shmidt-i) [inline src/Molecules/Input/Number/Number.types.ts]: `(x:{target: HTMLInputElement}) => void`

File: src/Molecules/Input/Number/Number.types.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+   * You need to specify width
+   * (better in pixels), because
+   * that will affect wrapping
+   * of the error/info text
+   * underneath
+   */
+  width?: string | number;
+
+  onStepUp?: (() => void) | ((value?: string) => void);
+  onStepDown?: (() => void) | ((value?: string) => void);
```
Comment (shmidt-i) [inline src/Molecules/Input/Number/Number.types.ts]: only `() => void` then

File: src/Molecules/Input/Number/adjustValue.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+
+const adjustValue = ({
+  originalValue,
+  step,
+  shouldIncrement,
+  intl,
+  min = -Infinity,
+  max = Infinity,
+}: adjustValueProps) => {
+  const getNumberOfDecimals = R.pipe(
```
Comment (shmidt-i) [inline src/Molecules/Input/Number/adjustValue.ts]: can move out of the function

File: src/Molecules/Input/Number/adjustValue.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    R.toString,
+    R.split('.'),
+    R.nth(1),
+    R.ifElse(R.isNil, R.always(0), R.length),
+  );
+  const numberOfDecimals = getNumberOfDecimals(step);
+  const multiplier = numberOfDecimals === 0 ? 1 : 10 ** numberOfDecimals; // 10^decimals
```
Comment (shmidt-i) [inline src/Molecules/Input/Number/adjustValue.ts]: `const multiplier = 10 ** numberOfDecimals`

**Comment by shmidt-i** [comment]
You can manually write value more than max and less than min
Add required to the text
Add autoFocus to the text
Disabled button still goes into active state (only css probably)
onChange should be triggered during every change

File: src/Atoms/Icon/__snapshots__/Icon.stories.storyshot [hunk 9]
```diff
@@ -552,6 +577,31 @@ exports[`Storyshots Atoms | Icon Available icons 1`] = `
...
+    >
+      Minus
+    </div>
+    <div>
+      <svg
+        aria-hidden="true"
+        className="c3"
+        focusable="false"
+        role="presentation"
+        size={5}
```
[the remark this answers is not in the document]
Comment (mjmonline) [reply src/Atoms/Icon/__snapshots__/Icon.stories.storyshot]: will fix in a separate PR

File: src/Molecules/Input/Number/Number.tsx (hunk 2 above)
Comment (mjmonline) [reply src/Molecules/Input/Number/Number.tsx]: fixed

File: src/Molecules/Input/Number/Number.tsx (hunk 4 above)
Comment (mjmonline) [reply src/Molecules/Input/Number/Number.tsx]: fixed

File: src/Molecules/Input/Number/adjustValue.ts (hunk 7 above)
Comment (mjmonline) [reply src/Molecules/Input/Number/adjustValue.ts]: fixed

File: src/Molecules/Input/Number/adjustValue.ts (hunk 8 above)
Comment (mjmonline) [reply src/Molecules/Input/Number/adjustValue.ts]: fixed

**Comment by mjmonline** [comment]
> You can manually write value more than max and less than min
> Add required to the text
> Add autoFocus to the text
> Disabled button still goes into active state (only css probably)
> onChange should be triggered during every change

**Pull request merged**